src/Tinfoil/Data/Key.hs

```diff
@@ -32,7 +32,15 @@ newtype SymmetricKey =
     unSymmetricKey :: ByteString
   } deriving (Eq, Generic)
 
-instance NFData SymmetricKey where rnf = genericRnf
+instance NFData SymmetricKey where
+  rnf = genericRnf
+
+-- | We use fixed valid Haskell tokens here to help prevent accidents such as
+-- inadvertantly logging secret keys.
+instance Show SymmetricKey where
+  showsPrec p _ =
+    showParen (p > 10) $
+      showString "SymmetricKey \"redacted\""
```

Comment:
@jystic can confirm, but I think this should be
```
showString "SymmetricKey " . showsPrec appPrec1 "redacted"
```

Can also use `appPrec` instead of `10` above